core/src/main/scala/com/yahoo/maha/utils/GetTotalRowsRequest.scala

```diff
@@ -0,0 +1,64 @@
+package com.yahoo.maha.utils
+
+import com.yahoo.maha.core._
+import com.yahoo.maha.core.query._
+import com.yahoo.maha.core.query.druid.DruidQueryGenerator
+import com.yahoo.maha.core.query.oracle.OracleQueryGenerator
+import com.yahoo.maha.core.registry.Registry
+import com.yahoo.maha.core.request.{Field, ReportingRequest}
+import grizzled.slf4j.Logging
+
+import scala.util.Try
+
+object GetTotalRowsRequest extends Logging {
+
+  def getTotalRowsRequest(request: ReportingRequest, pipeline: QueryPipeline) : Try[ReportingRequest] = {
+    //no filters except fk filters
+    Try {
+      require(
+        pipeline.bestDimCandidates.nonEmpty
+        , s"Invalid total rows request, no best dim candidates! : $request")
+
+      //force dim driven
+      //remove all fields except primary key
+      //remove all sorts
+      val primaryKeyAliasFields = pipeline.bestDimCandidates.map(dim => Field(dim.publicDim.primaryKeyByAlias, None, None)).toIndexedSeq
+      request.copy(
+        selectFields = primaryKeyAliasFields
+        , sortBy = IndexedSeq.empty
+        , includeRowCount = true
+        , forceDimensionDriven = true
+        , forceFactDriven = false
+        , paginationStartIndex = 0
+        , rowsPerPage = request.rowsPerPage
+      )
+    }
+  }
+
+  def getTotalRows(request: RequestModel, sourcePipeline: QueryPipeline, registry: Registry, queryContext: QueryExecutorContext)(implicit queryGeneratorRegistry: QueryGeneratorRegistry) : Try[Int] = {
+    Try {
```

Comment:
Consider changing this to a for {} yield {} since you have a bunch of try's within this try.